spec/jstests-spec.js

```diff
@@ -305,41 +304,126 @@ describe("Check correct url params", function() {
-  const cy = cytoscape({ 
+  const cy = cytoscape({
     elements: [
-    { data: { id: "A" } },
-    { data: { id: "B" } },
-    {
-      data: {
-        id: "AB",
-        source: "A",
-        target: "B"
-      }
-    }]
+      { data: { id: "A" } },
+      { data: { id: "B" } },
+      {
+        data: {
+          id: "AB",
+          source: "A",
+          target: "B"
+        }
+      }]
+  });
+
+  let numSelected;
+  let nodeError;
+  let query;
+  beforeEach(function() {
+    document.body.innerHTML = "";
+
+    numSelected = document.createElement("label");
+    numSelected.id = "num-selected";
+    document.body.appendChild(numSelected);
+
+    nodeError = document.createElement("label");
+    nodeError.id = "node-error";
+    document.body.appendChild(nodeError);
+
+    query = document.createElement("input");
+    query.id = "node-search";
+    document.body.appendChild(query);
+  });
+
+
+  it("should be a successful node search", function() {
+    query.value = "A";
+    const result = search(cy, "node", searchNode);
+
+    // should not display error message
+    expect(nodeError.innerText).toBe("");
+  });
+
+  it("should be an unsuccessful node search", function() {
+    query.value = "C";
+    const result = search(cy, "node", searchNode);
+
+    // should display error message
+    expect(nodeError.innerText).not.toBe("");
   });
 
-  it("should be a successful search", function() {
-    const result = searchNode(cy, "A");
+  it("should not search for a node at all", function() {
+    query.value = "";
+    const result = search(cy, "node", searchNode);
+
+    // should not display error message
+    expect(nodeError.innerText).toBe("");
+  });
+});
+
+describe("Token search", function() {
+  let numSelected;
+  let tokenError;
+  let query;
+  let cy;
+
+  beforeEach(function() {
+    document.body.innerHTML = `
+    <div id="cy"></div>`;
+    cy = cytoscape({
+      elements: [
+      ],
+      container: document.getElementById("cy"),
+    });
+    const nodeWithToken = {};
+    nodeWithToken["data"] = {};
+    nodeWithToken["data"]["id"] = "A";
+    nodeWithToken["data"]["tokens"] = ["a.js", "b.js", "c.js"];
+    cy.add(nodeWithToken);
+    const myNode = cy.nodes()[0];
+    initializeTippy(myNode);
+
+    numSelected = document.createElement("label");
+    numSelected.id = "num-selected";
+    document.body.appendChild(numSelected);
+
+    tokenError = document.createElement("label");
+    tokenError.id = "token-error";
+    document.body.appendChild(tokenError);
+
+    query = document.createElement("input");
+    query.id = "token-search";
+    document.body.appendChild(query);
+  });
+
+
+  it("should be a successful token search", function() {
+    query.value = "a.js";
+    const result = search(cy, "token", searchToken);
 
-    // search should find node
-    expect(result).toBe(true);
+    // error message should not be displayed
+    expect(tokenError.innerText).toBe("");
+    expect(result.length).toBe(1);
```

Comment:
Make assertions about the content as well, that it is what we expected it to be. Also make sure that all the edges and nodes are highlighted the way we want them to be.